### PR TITLE
Update cluster-api to v1.8.1

### LINKED ISSUE #1884: CAPI v1.8.0-beta.0 has been released and is ready for testing
CAPI v1.8.0-beta.0 has been released and is ready for testing.
Looking forward to your feedback before v1.8.0 release!

## For quick reference

- [CAPI v1.8.0-beta.0 release notes](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.8.0-beta.0)
- [Shortcut to CAPI git issues](https://github.com/kubernetes-sigs/cluster-api/issues)

## Following are the planned dates for the upcoming releases

CAPI v1.8.0 will be released on **Monday, 12th August 2024**.

More details of the upcoming schedule can be seen at [CAPI v1.8.0 release timeline](https://github.com/kubernetes-sigs/cluster-api/tree/main/docs/release/releases/release-1.8.md#timeline).

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
This PR updates
- cluster-api 1.7.4 -> 1.8.1
- golang 1.21 -> 1.22
- k8s 1.29 -> 1.30

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1884 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Update cluster-api to v1.8.1
```

### CODE REVIEW

**Comment by Karthik-K-N** [comment]
@Amulyam24 once you get sometime, Please help in fixing the test failures, We will try to get necessary things done by next week.

**Comment by Prajyot-Parab** [comment]
@Amulyam24 need to probably also update - https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/3b2125731bb9b77021128f222b165e53f853e6dd/cloudbuild.yaml#L7C5-L7C81
the updated image should have the go 1.22

**Comment by mkumatag** [comment]
but I think rc is already released!

**Comment by Amulyam24** [comment]
/hold cancel

PR is ready for review, PTAL!

**Comment by Amulyam24** [comment]
@mkumatag @Prajyot-Parab PTAL, PR is ready for review.

**Review by mkumatag — APPROVED** [review]
/lgtm

File: Makefile [hunk 1]
```diff
@@ -181,10 +174,9 @@ generate-manifests: $(CONTROLLER_GEN) ## Generate manifests e.g. CRD, RBAC etc.
 generate-go-conversions: $(CONVERSION_GEN) ## Generate conversions go code
 	$(MAKE) clean-generated-conversions SRC_DIRS="./api/v1beta1"
 	$(CONVERSION_GEN) \
-		--input-dirs=./api/v1beta1 \
```
[the remark this answers is not in the document]
Comment (Amulyam24) [reply Makefile:184]: yes, removed the deprecated flags and updated with new command format.

**Pull request merged**